spec/services/replication_transfer_updater_spec.rb

```diff
@@ -0,0 +1,266 @@
+        expect(described_class.new(their_transfer, nil).we_requested?).to be false
+      end
+    end
+  end
+
+
+  let(:updated) {
+    updated = ReplicationTransferAdapter.from_model(record).to_params_hash
+    updated.merge!(changes)
+  }
+
+  describe ReplicationTransferUpdater::CancelUpdate do
+    let(:update) { described_class.new(record, updated) }
+    let(:record) { Fabricate(:replication_transfer) }
+    let(:changes) { {cancelled: true, cancel_reason: "test"} }
+    describe "::matching_update?" do
+      it "matches old.cancelled == false, new.cancelled == true" do
+        expect(described_class.matching_update?(record, updated)).to be true
+      end
+    end
+    describe "#update" do
+      it "cancels the record" do
+        update.update
+        expect(record.cancelled).to be true
+      end
+      it "sets the cancel reason" do
+        update.update
+        expect(record.cancel_reason).to eql("test")
+      end
+      it "returns true on success" do
+        allow(record).to receive(:cancel!).and_return true
+        expect(update.update).to be true
+      end
+      it "returns false on failure" do
+        allow(record).to receive(:cancel!).and_return false
+        expect(update.update).to be false
+      end
+    end
+  end
+
+
+  describe ReplicationTransferUpdater::FixityValueUpdate do
+    let(:update) { described_class.new(record, updated) }
+    let(:fixity_alg) { Fabricate(:fixity_alg) }
+    let(:fixity_value) { "81234823749274982749273948723"}
+    let(:bag) {
+      bag = Fabricate(:bag)
+      bag.message_digests << Fabricate.build(:message_digest, fixity_alg: fixity_alg, value: fixity_value)
+      bag.save!
+      bag
+    }
+    let(:record) { Fabricate(:replication_transfer, bag: bag, fixity_alg: fixity_alg,
+      fixity_value: nil)}
+    let(:changes) { {fixity_value: fixity_value} }
+    describe "::matching_update?" do
+      it "matches old.fixity_value == nil, new.fixity_value != nil" do
+        expect(described_class.matching_update?(record, updated)).to be true
+      end
+    end
+    describe "#update" do
+      shared_examples "fixity_value_update_basics" do
+        it "updates the record" do
+          update.update
+          expect(record.fixity_value).to eql(changes[:fixity_value])
+        end
+        it "returns true on success" do
+          allow(record).to receive(:update).and_return true
+          expect(update.update).to be true
+        end
+        it "returns false on failure" do
+          allow(record).to receive(:update).and_return false
+          expect(update.update).to be false
+        end
+      end
+      context "we did not request this transfer" do
+        let(:record) { Fabricate(:replication_transfer, bag: bag, fixity_alg: fixity_alg,
+          fixity_value: nil)}
+        it_behaves_like "fixity_value_update_basics"
+        it "does not cancel the record" do
+          update.update
+          expect(record.cancelled).to be false
+        end
```

Comment:
This ends up being a check that a field doesn't change.  Pulling that thread leads to checking that every field we don't care about doesn't change.  That's not completely unreasonable to have, and could be accomplished by comparing against an expected record.  

However, I think that ends up duplicating the tests that are on the ReplicationTransfer model itself, as that's where the validation for this kind of stuff lives.  I think our tests become more fragile that way, too.  DIdn't seem like a tradeoff we wanted.  Do you disagree?